lib/Doctrine/ODM/MongoDB/Mapping/ReflectionEnumProperty.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Mapping;
+
+use BackedEnum;
+use ReflectionProperty;
+use ReturnTypeWillChange;
+use ValueError;
+
+use function assert;
+use function get_class;
+use function is_int;
+use function is_string;
+
+class ReflectionEnumProperty extends ReflectionProperty
+{
+    /** @var ReflectionProperty */
+    private $originalReflectionProperty;
+
+    /** @var class-string<BackedEnum> */
+    private $enumType;
+
+    /**
+     * @param class-string<BackedEnum> $enumType
+     */
+    public function __construct(ReflectionProperty $originalReflectionProperty, string $enumType)
+    {
+        $this->originalReflectionProperty = $originalReflectionProperty;
+        $this->enumType                   = $enumType;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param object|null $object
+     *
+     * @return int|string|null
+     */
+    #[ReturnTypeWillChange]
+    public function getValue($object = null)
+    {
+        if ($object === null) {
+            return null;
+        }
+
+        $enum = $this->originalReflectionProperty->getValue($object);
+
+        if ($enum === null) {
+            return null;
+        }
+
+        return $enum->value;
```

Comment:
Given this class is usable and actually loaded only in PHP 8.1 we should be safe to call
```suggestion
        return $this->originalReflectionProperty->getValue($object)?->value;
```
but feel free to ignore, probably there's a lot more places that would benefit from the approach in the future and by then we'll have a CS-rule that'd autofix everything.

Reply:
This `$enum` seems to become `null` when it's `null` in DB so I reverted that. See: https://github.com/doctrine/mongodb-odm/runs/5283674934?check_suite_focus=true

Reply:
That's because the `?` in `?->` got lost :)

Reply:
You are right @malarzm but after second thought, I decided to keep it pre-PHP8.1 compatible. This is in case someone accidentally imports this class in their code which would trow a really bad syntax error. I think we could revisit that topic once we decide to drop older PHP versions.